### PR TITLE
added zecko common java public

### PR DESCRIPTION
**Pull request by jai38**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
     <groupId>co.zecko.retailer</groupId>
     <artifactId>client</artifactId>
-    <version>0.0.5-RELEASE</version>
+    <version>0.0.6-SNAPSHOT</version>
```
Comment (akartik80) [inline pom.xml]: Use release versions

File: pom.xml [hunk 2]
```diff
@@ -186,5 +186,10 @@
             <artifactId>commons-lang3</artifactId>
             <version>3.12.0</version>
         </dependency>
+        <dependency>
+            <groupId>co.zecko.common.java</groupId>
+            <artifactId>client</artifactId>
+            <version>0.0.1-SNAPSHOT</version>
```
Comment (akartik80) [inline pom.xml]: release version

File: pom.xml [hunk 3]
```diff
@@ -186,5 +186,10 @@
             <artifactId>commons-lang3</artifactId>
             <version>3.12.0</version>
         </dependency>
+        <dependency>
+            <groupId>co.zecko.common.java</groupId>
```
Comment (akartik80) [inline pom.xml]:
```suggestion
            <groupId>co.zecko.common</groupId>
```

File: pom.xml [hunk 4]
```diff
@@ -186,5 +186,10 @@
...
+            <artifactId>client</artifactId>
```
Comment (akartik80) [inline pom.xml]:
```suggestion
            <artifactId>public</artifactId>
```

The private one can be: `co.zecko.common` and `private`

File: pom.xml [hunk 5]
```diff
@@ -186,5 +186,10 @@
             <artifactId>commons-lang3</artifactId>
```
Comment (akartik80) [inline pom.xml]: All these dependencies are not needed now as they will come with zecko-common-public

**Review by akartik80 — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (jai38) [reply pom.xml]: yes actually that i change in the final merge :)

File: pom.xml (hunk 5 above)
Comment (jai38) [reply pom.xml]: understood

File: src/main/java/co/zecko/retailer/common/pojo/Discount.java [hunk 6]
```diff
@@ -1,7 +1,7 @@
 package co.zecko.retailer.common.pojo;
 
 import co.zecko.retailer.common.enums.DiscountType;
-import co.zecko.retailer.common.pojo.base.BaseEntry;
+import co.zecko.common.java.pojo.BaseEntry;
```
Comment (akartik80) [inline src/main/java/co/zecko/retailer/common/pojo/Discount.java]:
```suggestion
import co.zecko.common.public.pojo.BaseEntry;
```

File: src/main/java/co/zecko/retailer/common/pojo/product/Product.java [hunk 7]
```diff
@@ -88,11 +88,15 @@ public class Product extends BaseEntry implements Serializable {
...
     public String getDiscountPercentage() {
         return DiscountUtil.getDiscountPercentage(getPrice(), getCompareAtPrice());
     }
 
-    public Boolean getIsAvailable () {
-        return totalInventory > 0;
+    public Boolean getIsAvailable() {
+        if(totalInventory == null) return null;
+        this.isAvailable =  totalInventory > 0;
+        return this.isAvailable;
```
Comment (akartik80) [inline src/main/java/co/zecko/retailer/common/pojo/product/Product.java:95]:
```suggestion
        return totalInventory == null ? null : totalInventory > 0;
```

File: src/main/java/co/zecko/retailer/common/pojo/Discount.java (hunk 6 above)
Comment (jai38) [reply src/main/java/co/zecko/retailer/common/pojo/Discount.java]: @akartik80 we can't use public keyword in import

File: pom.xml [hunk 8]
```diff
@@ -167,24 +167,9 @@
...
-        <dependency>
-            <groupId>com.fasterxml.jackson.core</groupId>
-            <artifactId>jackson-databind</artifactId>
-            <version>2.13.4.2</version>
-        </dependency>
-        <dependency>
-            <groupId>org.apache.commons</groupId>
-            <artifactId>commons-lang3</artifactId>
-            <version>3.12.0</version>
+            <groupId>co.zecko.commonPublic</groupId>
```
Comment (akartik80) [inline pom.xml]: This needs to be changed now, check main branch of common public

**Review by akartik80 — APPROVED** [review]

**Review by akartik80 — APPROVED** [review]

**Pull request merged**